src/test/java/io/strimzi/kafka/bridge/mqtt/mqtt/MqttBridgetIT.java

```diff
@@ -0,0 +1,179 @@
+ */
+public class MqttBridgetIT {
+
+    private static final Logger logger = LoggerFactory.getLogger(MqttBridgetIT.class);
+    private static final String BROKER_URI = "tcp://localhost:1883";
+    private static final String MQTT_TOPIC = "devices/bluetooth/type/audio/data";
+    private static final String KAFKA_TOPIC = "devices_bluetooth_data";
+    private static final String KAFKA_KEY = "devices_audio";
+    private static String CLIENT_ID = "myClientId";
+    private static MqttServer mqttBridge;
+    private static Thread prepareServerThread;
+    private static KafkaConsumerClient kafkaConsumerClient;
+
+
+    /**
+     * Start the kafka Cluster
+     * Start the MQTT bridge before all tests
+     */
+    @BeforeAll
+    public static void beforeAll() throws InterruptedException {
+        final Map<String, Object> config = new HashMap<>();
+        String kafkaBootstrapServers = "localhost:9092";
+        try (StrimziKafkaContainer container = new StrimziKafkaContainer()) {
+            container.start();
+            kafkaBootstrapServers = container.getBootstrapServers();
+        } catch (Exception e) {
+            logger.error("Error while starting the kafka cluster", e);
+        }
+
+        // prepare the configuration for the bridge
+        config.put(KafkaConfig.KAFKA_CONFIG_PREFIX + ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBootstrapServers);
+        config.put(BridgeConfig.BRIDGE_ID, "my-bridge");
+
+        // instantiate the kafka consumer client
+        kafkaConsumerClient = new KafkaConsumerClient(kafkaBootstrapServers);
+
+        // consumer client subscribes to the kafka topic
+        kafkaConsumerClient.subscribe(KAFKA_TOPIC);
+
+        // start the bridge server in a separate thread
+        prepareServerThread = new Thread(() -> {
+            try {
+                EventLoopGroup bossGroup = new NioEventLoopGroup();
+                EventLoopGroup workerGroup = new NioEventLoopGroup();
+
+                BridgeConfig bridgeConfig = BridgeConfig.fromMap(config);
+                String mappingRulesPath = Objects.requireNonNull(MqttBridgetIT.class.getClassLoader().getResource("mapping-rules-regex.json")).getPath();
+                MappingRulesLoader.getInstance().init(mappingRulesPath);
+
+                mqttBridge = new MqttServer(bridgeConfig, bossGroup, workerGroup, ChannelOption.SO_KEEPALIVE);
+                mqttBridge.start();
+            } catch (InterruptedException interruptedException) {
+                // in this case, we are interrupting the thread, so we don't need to do anything
+                logger.info("Shutting down the server");
+            } catch (Exception e) {
+                logger.error("Exception occurred", e);
+            }
+        });
+        prepareServerThread.start();
+        Thread.sleep(5000);
+    }
+
+    /**
+     * Create a new client id for each test randomly after each test
+     */
+    @AfterEach
+    public void afterEach() {
```

Comment:
Should be this a beforeEach? I would expect that before starting each test a new clientid is generated for such a test, or? Generating after is counterintuitive imho.

Reply:
Yes, it works! 
I was using afterEach because the clientId was already initialized.

What I am saying is somthing like
```java
 private String mqttClientId = "mqttClientId-" + new Random().nextInt();

@AfterEach
clientId += new Random().nextInt(10);
````
I think they are the same, aren't they?

Reply:
But the @BeforeEach seems to be simpler. I will go with that.